git-transport/src/client/blocking_io/file.rs

```diff
@@ -216,7 +231,8 @@ impl client::Transport for SpawnProcessOnDemand {
         if self.ssh_cmd.is_some() {
             cmd.args.push(service.as_str().into());
         }
-        cmd.args.push(self.path.to_os_str_lossy().into_owned());
+        cmd.args
+            .push(to_single_quoted(self.path.as_ref()).to_os_str_lossy().into_owned());
```

Comment:
`SpawnProcessOnDemand` isn't only used by `ssh`, but also when spawning processes locally.
This transformation should only happen when `ssh_cmd` is set.